src/errors.js

```diff
@@ -0,0 +1,17 @@
+class BaseError extends Error {
+	constructor(message) {
+		super(message);
+		this.name = this.constructor.name;
+		Error.captureStackTrace(this, this.constructor);
+	}
+}
+
+class UrlAlreadySubmittedError extends BaseError {
+	constructor(message) {
+		super(message || 'This url was already submitted previously.');
+	}
+}
+
+module.exports = {
+	UrlAlreadySubmittedError,
+};
```

Comment:
Did you really need to create a `BaseError` class here or could you have done:

```
 class UrlAlreadySubmittedError extends Error {
	constructor(message) {
		super(message);
		this.name = this.constructor.name;
		Error.captureStackTrace(this, this.constructor);
	}

	constructor(message) {
		super(message || 'This url was already submitted previously.');
	}
}

```

I think `BaseError` would be useful if it was used multiple times.

Reply:
Yep you're right, it's just easier this way as next error needed, we can keep extending our `BaseError` and not have to re-do this setup for each error.

Reply:
Oh ok so you already know there will be another error needed. Gotcha.